### PR TITLE
HW7: NoSQL Databases. ODM

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/models/review.js [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
-    this.productId = productId;
-    this.author = author;
-    this.text = text;
-  }
-}
+import mongoose from 'mongoose';
+
+const reviewSchema = mongoose.Schema({
+  _id: { type: String },
+  productId: { type: String, required: true },
```
Comment (boccob) [inline src/models/review.js:5]: Btw it's possible to specify relations in mongoose:
http://mongoosejs.com/docs/populate.html

File: src/helpers/routeHelper.js [hunk 2]
```diff
@@ -1,47 +1,131 @@
...
+  static async getProducts(req, res) {
+    const products = await ProductController.getProducts();
+    ExpressMiddleware.sendResponse(res, 200, products);
   }
 
-  static productNotFound(req, res) {
-    res.status(404);
-    res.end(`${config.product_not_found}: ${req.params.id}`);
+  static async deleteProduct(req, res) {
+    const product = await ProductController.deleteProduct(req.params.id);
```
Comment (boccob) [inline src/helpers/routeHelper.js]: It'd be good to remove reviews connected with this product.
Another option - make reviews a part of products collection.

File: src/index.js [hunk 3]
```diff
@@ -1,11 +1,54 @@
...
-app.listen(port, () => logger.info(`App ${config.server_listening} ${port}`));
+
+initLocale();
+initPassport();
+
+(async () => {
+  try {
+    await connectors.MONGO.connect();
+
+    CityController.addCity('Brest', 'Belarus', false, 52.097621, 23.734050);
```
Comment (boccob) [inline src/index.js]: Pretty nice try but when you run it the 2nd time you add the same data again (+ there are `E11000 duplicate key error collection` errors).
I'd move it to separate file and add to npm scripts.

File: src/models/review.js [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
-    this.id = String(id);
-    this.productId = productId;
-    this.author = author;
-    this.text = text;
-  }
-}
+import mongoose from 'mongoose';
+
+const reviewSchema = mongoose.Schema({
+  _id: { type: String },
```
Comment (boccob) [inline src/models/review.js]: Why not Schema.Types.ObjectId? And I think you missed `unique: true`.

**Review by boccob — APPROVED** [review]

File: src/models/review.js (hunk 4 above)
Comment (ghost) [reply src/models/review.js]: Yeah, I might miss this one day

File: src/models/review.js (hunk 1 above)
Comment (ghost) [reply src/models/review.js:5]: Thanks

File: src/helpers/routeHelper.js (hunk 2 above)
Comment (ghost) [reply src/helpers/routeHelper.js]: Fixed

**Pull request merged**